apps/ledger-live-mobile/src/components/CustomImage/imageUtils.ts

```diff
@@ -0,0 +1,144 @@
+import { Image } from "react-native";
+import RNFetchBlob, { FetchBlobResponse, StatefulPromise } from "rn-fetch-blob";
+import * as ImagePicker from "expo-image-picker";
+import { ImageDimensions, ImageFileUri, ImageUrl } from "./types";
+import {
+  ImageDownloadError,
+  ImageLoadFromGalleryError,
+  ImageMetadataLoadingError,
+  ImageTooLargeError,
+} from "./errors";
+
+export async function importImageFromPhoneGallery(): Promise<
+  (ImageFileUri & Partial<ImageDimensions>) | null
+> {
+  try {
+    const result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      allowsEditing: false,
+      quality: 1,
+      base64: false,
+    });
+    if (result.cancelled) {
+      return null;
+    }
+    const { uri, width, height } = result;
+    if (uri) {
+      return {
+        width,
+        height,
+        imageFileUri: uri,
+      };
+    }
+    throw new Error("uri is falsy");
+  } catch (e) {
+    console.error(e);
+    throw new ImageLoadFromGalleryError();
+  }
+}
+
+export function downloadImageToFileWithCancellable({
+  imageUrl,
+}: ImageUrl): [Promise<ImageFileUri>, StatefulPromise<FetchBlobResponse>] {
+  const downloadTask = RNFetchBlob.config({ fileCache: true }).fetch(
+    "GET",
+    imageUrl,
+  );
+  return [
+    downloadTask
+      .then(res => ({ imageFileUri: "file://" + res.path() }))
+      .catch(() => {
+        throw new ImageDownloadError();
+      }),
+    downloadTask,
+  ];
+}
+
+export async function downloadImageToFile({
+  imageUrl,
+}: ImageUrl): Promise<ImageFileUri> {
+  return downloadImageToFileWithCancellable({ imageUrl })[0];
+}
+
+export async function loadImageToFileWithDimensions(
+  source: Partial<ImageUrl> & Partial<ImageFileUri> & Partial<ImageDimensions>,
```

Comment:
Is `source: ImageUrl | (ImageFileUri & Partial<ImageDimensions>)` more appropriate ?